app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -16,7 +16,27 @@ def registration_args
   end
 
   def insecure
-    registration_params['insecure'] ? '--insecure' : ''
+    if registration_params['insecure']
+      if download_utility == 'curl'
+        ' --insecure'
+      else
+        ' --no-check-certificate'
+      end
+    else
+      ''
+    end
+  end
+
+  def application
+    if download_utility == 'curl'
+      'curl -sS'
+    else
+      'wget --no-verbose -O-'
+    end
+  end
+
+  def download_utility
+    registration_params['download_utility']
```

Comment:
What if the parameter is not set? By default the value should be a 'curl'

Reply:
Added `curl` as default